.github/workflows/ci.yml

```
@@ -461,7 +462,7 @@ jobs:
         run : git config --global --add safe.directory '*'
 
       - name: Checkout
-        uses: actions/checkout@v3 # TODO https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/
+        uses: actions/checkout@v3
```

Doesn't the following fix the original problem? (Please note that I did not test.)
```suggestion
        uses: actions/checkout@v4
```

---

Ah, sorry. That's an ancient Linux distro that would probably require some tweaking to get Node 20 running on it in the first place.
Like https://tecadmin.net/install-latest-nodejs-and-npm-on-centos/ pointing to https://rpm.nodesource.com/